Controllers films users

KYazykov opened the pull request.

KYazykov committed:
"Начальный коммит"

KYazykov committed:
"Начальный коммит"

KYazykov committed:
"Начальный коммит"

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,88 @@
+        if (!films.isEmpty()) {
+            for (Film film1 : films.values()) {
+                if (film1.getName().equals(film.getName())) {
+                    log.info("Выявлена ошибка валидации, фильм с таким названием уже зарегистрирован");
```
Есть фильмы с одинаковыми названиями. Думаю, что такая проверка лишняя. 
Если сравниваются имена или названия, то лучше приводить к одному регистру, например, нижнему.

KYazykov replied:
Александр, здравствуйте! 
Удалил лишнюю проверку на название. Привел проверку на емейл в User к нижнему регистру.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -0,0 +1,88 @@
+
+    @PutMapping
+    public Film updateFilm(@RequestBody Film film) {
+        if (film.getName() == null || film.getName().isBlank()) {
```
Все эти проверки очень похожи на соответствующие проверки из метода Add. Есть предложение провести небольшой  рефакторинг и сделать один общий метод проверки.

KYazykov replied:
Создал общий метод проверки в каждом классе.

avfyodorov commented on src/main/java/ru/yandex/practicum/filmorate/model/Film.java:
```diff
@@ -0,0 +1,18 @@
+@Builder
+public class Film {
+
+    private String name;
```
Можно добавить ограничения на размер (для отдельных полей).
@SiZe(min = 1, max = 200)
Для этих полей можно было бы добавить аннотации:
@notblank
@SiZe(min = 1, max = 200)
@NotNull
@min(1)

KYazykov replied:
Добавил аннотации по возможности, но проверку в методах не менял/удалял.

avfyodorov reviewed (commented):
Добрый вечер, Константин! 

Работа получилась хорошей, правда есть пара моментов, которые хорошо было бы уточнить. Я не стал писать одинаковые комментарии, для фильмов и пользователей, когда будете править, посмотрите пожалуйста все классы.

KYazykov committed:
refactor "Сделать общий метод для проверки для каждого класса"
add "Добавить аннотации и зависимость к ним"
delete "Удалить проверку на наименование фильма"

avfyodorov reviewed (commented):
Добрый день, Константин!

Замечаний нет, работа принята, всё отлично.
Успехов!
